ops/scripts/pipeline/azure-deploy.sh

```diff
@@ -125,12 +159,97 @@ while [[ $# -gt 0 ]]; do
         ;;
     esac
 done
+if [[ -z "${deployment_file}" ]]; then
+    echo "Error: Missing deployment file"
+    exit 11
+fi
+
+if [ ! -f "${deployment_file}" ]; then
+    echo "Error: File (${deployment_file}) does not exist."
+    exit 12
+fi
 
-validation_func "${app_rg}" "${deployment_file}" "${deployment_parameters}"
+if [[ -z "${deployment_parameters}" ]]; then
+    echo "Error: Missing deployment parameters"
+    exit 13
+fi
+if [[ -z "${app_rg}" ]]; then
+    echo "Error: Missing default resource group"
+    exit 10
+fi
+if [[ -z "${app_name}" ]]; then
+    echo "Error: Missing appName"
+    exit 10
+fi
+if [[ -z "${network_resource_group}" ]]; then
+    echo "Error: Missing Network resource group"
+    exit 10
+fi
+if [[ -z "${vnet_name}" ]]; then
+    echo "Error: Missing Vnet Name"
+    exit 10
+fi
+if [[ -z "${analytics_workspace_id}" ]]; then
+    echo "Error: Missing analytics workspace id"
+    exit 10
+fi
+if [[ -z "${keyvault_app_config_id}" ]]; then
+    echo "Error: Missing KV App config ID Name"
+    exit 10
+fi
+if [[ -z "${cosmos_id_name}" ]]; then
+    echo "Error: Missing Cosmos Managed id name"
+    exit 10
+fi
+if [[ -z "${cams_react_select_hash}" ]]; then
+    echo "Error: Missing camsReactSelectHash"
+    exit 10
+fi
+if [[ -z "${ustp_issue_collector_hash}" ]]; then
+    echo "Error: Missing ustpIssueCollectorHash"
+    exit 10
+fi
+if [[ -z "${deploy_vnet}" ]]; then
+    echo "Error: Missing deployVnet"
+    exit 10
+fi
+if [[ -z "${webapp_plan_type}" ]]; then
+    echo "Error: Missing webappPlanType"
+    exit 10
+fi
+if [[ -z "${function_plan_type}" ]]; then
+    echo "Error: Missing functionPlanType"
+    exit 10
+fi
+if [[ -z "${deploy_functions}" ]]; then
+    echo "Error: Missing deployFunctions"
+    exit 10
+fi
+if [[ -z "${deploy_webapp}" ]]; then
+    echo "Error: Missing deployWebapp"
+    exit 10
+fi
+if [[ -z "${action_group_name}" && "${create_alerts}" == true ]]; then
+    echo "Create Alerts: ${create_alerts} but no actionGroupName supplied"
+    exit 10
+fi
```

Comment:
This seems extremely verbose. Could we implement something else to check for required variables?

Reply:
Intent to handle this better in the second iteration of bicep improvements